sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -195,11 +201,16 @@ export class ServiceClient {
 
         authPolicyFactory = wrappedPolicyFactory();
       } else if (credentials && typeof credentials.signRequest === "function") {
+        logger.info("ServiceClient: creating signing policy from provided credentials");
         authPolicyFactory = signingPolicy(credentials);
       } else if (credentials !== undefined) {
+        logger.warning(
```

Comment:
I think this is unnecessary since an error will be coming out (and, really, this error is unlikely to be something users see but for us, in which case the logging isn't super useful).